src/components/cards/CreatePool/CreateActions.vue

```diff
@@ -0,0 +1,250 @@
+  confirming: false,
+  confirmed: false,
+  confirmedAt: ''
+};
+
+/**
+ * PROPS & EMITS
+ */
+const props = defineProps<Props>();
+
+const emit = defineEmits<{
+  (e: 'success', value: TransactionReceipt): void;
+}>();
+
+/**
+ * STATE
+ */
+const currentActionIndex = ref(0);
+const joinPoolState = reactive<ActionState>({ ...defaultActionStateValues });
+const createPoolState = reactive<ActionState>({ ...defaultActionStateValues });
+
+const joinPoolStepState = generateStepState(joinPoolState);
+const createPoolStepState = generateStepState(createPoolState);
+
+/**
+ * COMPOSABLES
+ */
+const route = useRoute();
+const { t } = useI18n();
+const { networkConfig } = useConfig();
+const { account, getProvider, explorerLinks } = useWeb3();
+const { addTransaction } = useTransactions();
+const { txListener, getTxConfirmedAt } = useEthers();
+const { parseError } = useTranasactionErrors();
+const { fiatTotalLabel } = useTokenFiatMath(
+  props.tokenAddresses,
+  props.amounts
+);
+const { allApproved, tokenApprovalActions } = useTokenApprovalActions(
+  props.tokenAddresses,
+  props.amounts,
+  currentActionIndex
+);
+const { createPool, joinPool } = usePoolCreation();
+
+/**
+ * COMPUTED
+ */
+
+const actions = computed((): Action[] => [
+  {
+    label: t('createpool'),
+    loadingLabel: t('confirming'),
+    pending: createPoolState.init || createPoolState.confirming,
+    promise: submit.bind(null, createPool, createPoolState),
+    step: {
+      tooltip: t('createPoolTooltip'),
+      state: createPoolStepState.value
+    }
+  },
+  ...tokenApprovalActions.value,
+  {
+    label: t('joinpool'),
+    loadingLabel: joinPoolState.init
+      ? t('investment.preview.loadingLabel.investment')
+      : t('confirming'),
+    pending: joinPoolState.init || joinPoolState.confirming,
+    promise: submit.bind(null, joinPool, joinPoolState),
+    step: {
+      tooltip: t('investmentTooltip'),
+      state: joinPoolStepState.value
+    }
+  }
+]);
+
+const currentAction = computed(
+  (): Action => actions.value[currentActionIndex.value]
+);
+
+const steps = computed((): Step[] => actions.value.map(action => action.step));
+
+const explorerLink = computed((): string =>
+  joinPoolState.receipt
+    ? explorerLinks.txLink(joinPoolState.receipt.transactionHash)
+    : ''
+);
+
+/**
+ * METHODS
+ */
+
+function generateStepState(actionState: ActionState): ComputedRef<StepState> {
+  return computed(
+    (): StepState => {
+      if (tokenApprovalActions.value.length > 0 && !allApproved.value) {
+        return StepState.Todo;
+      } else if (actionState.confirming) return StepState.Pending;
+      else if (actionState.init) return StepState.WalletOpen;
+      else if (actionState.confirmed) return StepState.Success;
+      return StepState.Active;
+    }
+  );
+}
+
+async function submit(
+  action: () => Promise<TransactionResponse>,
+  state: ActionState
+): Promise<void> {
+  try {
+    state.init = true;
+
+    const tx = await action();
```

Comment:
Are we not adding these transaction to the global notifcations?

Reply:
They're being added in the action itself. It handles the global notifications, while this section handles the ActionSteps.